src/components/timeline/LayerDiscrete.svelte

```diff
@@ -562,30 +750,108 @@
       if (directive && showDirectives && directiveInView(directive, viewTimeRange)) {
         newItem = { ...newItem, directive, directiveStartX: xScaleView(directive.start_time_ms ?? 0) };
       }
+
+      // should never occur at the same time as the above. as such, not written as newItem = {...newItem, ...}
+      if (externalEvent && externalEventInView(externalEvent, viewTimeRange)) {
+        newItem = { externalEvent, externalEventStartX: xScaleView(externalEvent.start_ms) };
+      }
       if (newItem) {
         itemsToDraw.push(newItem);
       }
     });
 
-    itemsToDraw.forEach(({ directive, directiveStartX, span, spanStartX }, i) => {
+    // sort the items (this is crucial to ensuring drawlabel funcitonality works correctly)
```

Comment:
Directives and spans _should_ already be sorted. Instead of doing a sort here could we just pre-sort events before passing them into the layer/timeline?

Reply:
I see; they appear to be sorted in `Row.svelte` when creating `filteredActivityDirectives`/`filteredSpans`; this behavior has been replicated to `filteredExternalEventsByTime` and removed from `LayerDiscrete.svelte` in [remove unnecessary sorting in `LayerDiscrete.svelte`](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/e74e4c69c7a06f642e5fd2589ef5422abaedd10e).